### ISSUE TITLE
location of mower on map

### ISSUE THREAD

**Issue by goosst** [EinEinfach/CaSSAndRA#116]
Hello 

There seems to be bug which seems quite repetitive in my configuration.

The robot is in reality positioned at the edge of the map (the gps has a fix). However in cassandra it gets reported somewhere inside the map. If I let it mow, it also goes outside the field ... .
The result does not change when rebooting cassandra and sunray through the cassandra app (pressing the R).
![image](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/5056fb7c-c583-4b39-911a-c6933f347e7b)

After fully rebooting with `sudo systemctl restart cassandra`, it does get reported to the correct location.
![image](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/f328e644-a6c3-4fb3-883c-c12271cfcefe)

Over mqtt, the first time (wrong map), the following is reported:
`/ardumower/Ardumower/state {"battery_voltage":25.78,"position":{"x":-0.39,"y":-1.27,"delta":0.41,"solution":2,"age":1.12,"accuracy":0.02,"visible_satellites":43,"visible_satellites_dgps":42,"mow_point_index":0},"target":{"x":0,"y":0},"job":0,"sensor":0,"amps":0,"map_crc":-25170}`

The second time (correct map), the following is reported:
`/ardumower/Ardumower/state {"battery_voltage":25.73,"position":{"x":2.3,"y":2.56,"delta":0.42,"solution":2,"age":0.1,"accuracy":0.02,"visible_satellites":43,"visible_satellites_dgps":41,"mow_point_index":0},"target":{"x":0,"y":0},"job":0,"sensor":0,"amps":0,"map_crc":-25170}`

I'm using NTRIP, absolute gps coordinates.

**Comment by EinEinfach** [comment]
Which position is reported by using sunray app? Pls share your Cassandra log entries. After rebooting and wrong position and after full reboot and right position

**Comment by goosst** [comment]
so I started the app with this command: 
`./app.py --app_log_file_level DEBUG --app_log_level DEBUG --server_log_level DEBUG`

[cassandratransfer.log](https://github.com/EinEinfach/CaSSAndRA/files/14409517/cassandratransfer.log)

1. So I cut the power, position the robot somewhere, wait for a gps fix, load a map. Mower is at the wrong location: ![Screenshot from 2024-02-26 19-42-29](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/6d58db55-e59d-4feb-8ec0-83b7f070281c)
2. Then I restart the app and mower is at the correct location ![Screenshot from 2024-02-26 19-43-04](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/785bd25f-d4e4-4301-aa49-7303d08cdfe1)

**Comment by EinEinfach** [comment]
Please set ENABLE_PASS false in your sunray config.h an recompile the firmware, then repeat the procedure. It very hard to follow your Logfile with encrypted communication. And pls set server_log level to error

**Comment by goosst** [comment]
ok, I'll do that in a few days

**Comment by goosst** [comment]
So I've played a bit more with the robot.
Once I've updated the software with ENABLE_PASS disabled, the occurence of this issue seems far less (i.e. it does not seem repetitive anymore).
Unfortunately I had it still one time today when starting up the mower after cassandra was running for a while. But I was not running cassandra in debug mode ... so no log. 
(Running it by default in debug mode created some strange state where the log was not updating anymore etc. .)

So I guess, less useful update than expected ... :).

**Comment by goosst** [comment]
1) started mower in his docking station
2) carried it to his mowing location --> still reports as docked
3) restart cassandra --> still reports as docked
4) restart robot --> robot is positioned in wrong location despite gps fix  (screenshot 1)
5) restart cassandra --> robot is in correct location (screenshot 2)

![Screenshot from 2024-03-04 18-26-20](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/fc4a2455-767e-42c1-8f7e-7df4fd7201af)

![Screenshot from 2024-03-04 18-26-47](https://github.com/EinEinfach/CaSSAndRA/assets/8907127/efb4ba6c-bb9e-4b3a-bffe-07ed608cda48)


logs are attached:
[cassandratransfer2.log](https://github.com/EinEinfach/CaSSAndRA/files/14485434/cassandratransfer2.log)
[cassandratransfer1.log](https://github.com/EinEinfach/CaSSAndRA/files/14485435/cassandratransfer1.log)
[cassandratransfer.log](https://github.com/EinEinfach/CaSSAndRA/files/14485436/cassandratransfer.log)

**Comment by EinEinfach** [comment]
I assume it's something wrong with your sunray fw or sunray fw configuration:
1. Your robot reports docked state even if it not in charging station (maybe a hardware issue)
2. Your robot change his position after receiving position mode command. It's ok, if it is done first time, but not in productive enviroment:
2024-03-04 17:26:31 INFO RX: S,26.22,**-4.60**,**-3.52**,0.39,2,0,0,0.83,0,0.00,0.00,0.02,35,0.00,35,0,0.00,-1,0,0x16

2024-03-04 17:26:31 DEBUG Recieved checksum:0x16 Calced checksum: 0x16
2024-03-04 17:26:31 DEBUG Backend: Calc data from state data frame
2024-03-04 17:26:31 DEBUG AT+P,1,3.919431,50.903403 will be send to the rover
2024-03-04 17:26:31 DEBUG Calced checksumme: 0x27
**2024-03-04 17:26:31 INFO TX AT+P,1,3.919431,50.903403,0x27**
2024-03-04 17:26:31 DEBUG Starting new HTTP connection (1): 192.168.0.175:80
2024-03-04 17:26:32 DEBUG http://192.168.0.175:80 "POST / HTTP/1.1" 200 8
2024-03-04 17:26:32 INFO RX: P,0x50

2024-03-04 17:26:33 INFO Performing get state http-request
2024-03-04 17:26:33 DEBUG Calced checksumme: 0x13
2024-03-04 17:26:33 INFO TX: AT+S,0x13
2024-03-04 17:26:33 DEBUG Starting new HTTP connection (1): 192.168.0.175:80
2024-03-04 17:26:34 DEBUG http://192.168.0.175:80 "POST / HTTP/1.1" 200 81
2024-03-04 17:26:34 DEBUG Status code: 200
2024-03-04 17:26:34 INFO RX: S,26.22,**27.55**,**6.84**,0.39,2,0,0,0.59,0,0.00,0.00,0.02,35,0.00,35,0,0.00,-1,0,0x00

For me it's not clear, why. Maybe is there another client online... 
Sorry but I can't help you
